### ISSUE TITLE
roms built with latest uxn aren't working

### ISSUE THREAD

**Issue by la-ninpre** [aduros/webuxn#2]
i tried to run webuxn locally, then i loaded piano.rom which i just compiled with latest version of uxn and it is just a black screen.

for comparison, if i take piano.rom from this repo, it runs fine.

i am not at all an expert in wasm and stuff, but maybe there was some incompatible change in upstream uxn?

~~both roms run fine with uxnemu though...~~

UPD: tested old roms with current uxnemu, and they aren't working.

**Comment by aduros** [comment]
I think there have been upstream breaking changes to uxn. It's hopefully just a matter of pulling in the latest C sources (uxn-fast.c, ppu.c, etc) from upstream though.

If you get it working, can you please open a PR?

cc @neauoire who probably knows more about it.

**Comment by neauoire** [comment]
yup, it's just a matter of pulling the latest changes, it should work right off :)

**Comment by la-ninpre** [comment]
i haven't get it working yet. i'll try, though. and if it'll be successfull, i'll make a pull request.

oh, and also, kinda aside question: on which platform and using what tools webuxn is supposed to be built? i'm asking because the build fails for me in archlinux…

**Comment by aduros** [comment]
You'll need at least `emcc` from emscripten, and nodejs to build the JS side.

**Comment by la-ninpre** [comment]
hmm, maybe i'm doing something wrong, but when i put new uxn-fast.c, apu.{h,c}, ppu.{h,c} and uxn.h it doesn't work at all. it seems that a lot of functions now have different names and even different prototypes. like, really, there are a lot of changes and that needs a bit of research, because, for example, there are no Layer struct in new uxn, there are no WIDTH and HEIGHT defines and so on…

**Comment by neauoire** [comment]
Ah damn, that's true, the Layer is just a number(`0-1`) now.
The `WIDTH` and `HEIGHT` defines are still there?
https://git.sr.ht/~rabbits/uxn/tree/main/item/src/uxnemu.c#L24

**Comment by la-ninpre** [comment]
hmm, `WIDTH`' and `HEIGHT` are there, but they are defined in `uxn.c` which is not used in webuxn. there is only `uxn-fast.c`.
